x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/parser/LogicalPlanBuilder.java

```diff
@@ -102,7 +102,13 @@ public Object visitStatement(StatementContext ctx) {
             if (ctx.pipe().size() > 0) {
                 throw new ParsingException(source(ctx.pipe().get(0)), "Samples do not support pipes yet");
```

Comment:
Shouldn't support for pipes now also be possible since limit and offset are both supported?

Reply:
No, because the limit/size comes from a request parameter, not from the query itself (through the query parser).

Reply:
but it works with `sequence`, doesn't it? why would `sample` be different in this regard?

Reply:
With `sequence` you have `| head 5 | tail 3 | head 1`. How do you do this with `sample`s?

Reply:
I see HEAD and TAIL as very natural operators for sequences, that have well defined semantics defined by their order (eg. `TAIL 10`: give me last ten - more recent - events). 
Samples, on the other hand, do not have a natural order: "last ten samples" is not a clear concept by itself.
In this context, LIMIT/OFFSET is much more similar to the SQL equivalent, where you just need a way to retrieve the results in batches or to show them in a UI in pages.
I agree with @Luegg on the fact that we could support pipes here eventually, but we probably need some slightly different syntax, eg `| LIMIT 10 OFFSET 20`; or reuse HEAD with `| OFFSET 20 | HEAD 10`, though it seems less natural in this context.

Reply:
@Luegg, as @luigidellaquila well put this, re-using `HEAD` and `TAIL` doesn't reflect the nature of `sample`s which do not have a chronological ordering concept, like `sequence`s do. The only type of ordering that make sense is the one on the join keys, but at that point the simple concepts of `HEAD` and `TAIL` are not enough to express the multitude of ordering options available (order by first join key asc, order by the second join key desc, order by the third join key asc and so on; or order by all keys asc, or all keys desc, but at that point one would ask the question why cannot I order by each key individually etc). At this point in the evolution of `sample`s, I believe having a circuit breaker and a simple way of limiting (without computing all the samples) the number of results returned, is enough. With time, and more input from users, we can evaluate our options in providing something more advanced and more granular in terms of limiting the number of results and, at the same time, sorting the results.

Reply:
@Luegg I agree with you, tail is definitely not straight forward at this stage.
Also for a proper implementation of pipes on samples (and limit/offset in particular), I think we still need to define a few small details that do not really fit in the scope of this PR. 
For this specific one, I'd suggest to stick to the original goal, but keeping a door open for further improvements.